elifepubmed/generate.py

```diff
@@ -601,6 +601,45 @@ def set_grants(parent, poa_article):
                 set_object(parent, "grant", params)
 
 
+def dataset_assigning_authority(assigning_authority, uri):
+    """precise assigning_authority value considering the uri in some cases"""
+    authority_map = OrderedDict([('NCBI', [])])
+    # tuple in form: uri hint, new assigning authority
+    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/geo', 'NCBI:geo'))
+    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/projects/gap', 'NCBI:dbgap'))
+    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/nuccore', 'NCBI:nucleotide'))
+    authority_map['NCBI'].append(('www.ncbi.nlm.nih.gov/sra', 'NCBI:sra'))
```

Comment:
`authority_map` is a constant so I would extract it from this function so it's not added to memory every time the function is called.

Reply:
A good place for it is outside, agreed!